### PR TITLE
xtest

### PR DESCRIPTION
**Pull request by Urhengulas**
This PR transfers the non-static CI steps into the command `cargo xtest`.

* `RUSTFLAGS='--deny warnings' cargo build` is done by clippy, since we specify `-- --deny warnings`
* the action `actions-rs/clippy-check@v1` is kept, since it gives pretty inline comments 

## Why `xtest` and not `xtask`?
Becasue it is mainly running `cargo test`, and just installs `flip-link` before it.

I started off with `xtask`, since this is the common name. But if feels a bit weird to just do `cargo xtask`, without sepcifying a subcommand, so I thought about `cargo xtask test`, but this is also weird, if there are no other subcomands.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,11 +88,18 @@ rustflags = [
...
-NOTE that if you were using GNU `ld` or GNU `gcc` to link your program then this
-won't work. Support for other linkers is being tracked in [issue #1]
+NOTE that if you were using GNU `ld` or GNU `gcc` to link your program then this won't work. Support for other linkers is being tracked in [issue #1].
 
 [issue #1]: https://github.com/knurling-rs/flip-link/issues/1
 
+## Testing
+
+Our CI enforces various checks. You can run them locally to make sure your PR will pass the CI:
+* `cargo fmt -- --check`
```
Comment (justahero) [inline README.md:98]: Very minor nitpick (& probalby irrelevant), the command executed is `cargo fmt --all -- --check`.

**Review by justahero — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (Urhengulas) [reply README.md:98]:
```suggestion
* `cargo fmt --all -- --check`
```

**Pull request merged**